dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java

```diff
@@ -0,0 +1,719 @@
+        final HttpServletRequest request = mock(HttpServletRequest.class);
+        final HttpSession session = mock(HttpSession.class);
+
+        setAllParametersAndAttributes(request, session);
+        when(request.getParameter("host_id")).thenReturn(host.getIdentifier());
+
+        final User user = createBackendUser(host);
+
+        final HostWebAPIImpl hostWebAPI = new HostWebAPIImpl();
+
+        final Host currentHost = hostWebAPI.getCurrentHost(request, user);
+
+        assertEquals(host.getIdentifier(), currentHost.getIdentifier());
+        verify(request).setAttribute(WebKeys.CURRENT_HOST, currentHost);
+        verify(session).setAttribute(WebKeys.CURRENT_HOST, currentHost);
+    }
+
+    /**
+     * Method to test: {@link HostWebAPIImpl#getCurrentHost(HttpServletRequest, User)} (HttpServletRequest)}
+     * When: request host_id parameter is set and the user is not a backend user
+     * Should: Host parameter should by taken
+     */
+    @Test
+    public void useHostIDRequestAsCurrentHostNotBackendUser() throws DotDataException, DotSecurityException {
+        final Host host = new SiteDataGen().nextPersisted();
+        final Host anotherHost = new SiteDataGen().nextPersisted();
+        final HttpServletRequest request = mock(HttpServletRequest.class);
+        final HttpSession session = mock(HttpSession.class);
+
+        setAllParametersAndAttributes(request, session, anotherHost);
+        when(request.getParameter("host_id")).thenReturn(host.getIdentifier());
+
+        final User user = createUser(host);
+
+        final HostWebAPIImpl hostWebAPI = new HostWebAPIImpl();
+
+        final Host currentHost = hostWebAPI.getCurrentHost(request, user);
+
+        assertEquals(anotherHost.getIdentifier(), currentHost.getIdentifier());
+        verify(request).setAttribute(WebKeys.CURRENT_HOST, currentHost);
+        verify(session, never()).setAttribute(anyString(), anyObject());
+    }
+
+    /**
+     * Method to test: {@link HostWebAPIImpl#getCurrentHost(HttpServletRequest, User)} (HttpServletRequest)}
+     * When: request host_id parameter is set and the user is a backend user but don't have permission
+     * Should: throw a {@link DotSecurityException}
+     */
+    @Test
+    public void useHostIdAndTheUserDontHavePermission() throws DotDataException, DotSecurityException {
+        final Host host = new SiteDataGen().nextPersisted();
+        final HttpServletRequest request = mock(HttpServletRequest.class);
+        final HttpSession session = mock(HttpSession.class);
+
+        setAllParametersAndAttributes(request, session);
+        when(request.getParameter("host_id")).thenReturn(host.getIdentifier());
+
+        final User user = createBackendUser();
+
+        final HostWebAPIImpl hostWebAPI = new HostWebAPIImpl();
+
+        try {
+            hostWebAPI.getCurrentHost(request, user);
+            throw new RuntimeException("should throw a DotSecurityException");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal "should throw a DotSecurityException" appears 5 times in this file; the first occurrence is on line 105](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)